central/reports/scheduler/v2/reportgenerator/report_gen_impl.go

```diff
@@ -155,7 +155,11 @@ func (rg *reportGeneratorImpl) generateReportAndNotify(req *ReportRequest) error
 	}
 	switch req.ReportSnapshot.ReportStatus.ReportNotificationMethod {
 	case storage.ReportStatus_DOWNLOAD:
-		if err = rg.saveReportData(req.ReportSnapshot.GetReportConfigurationId(),
+		reportConfigId := req.ReportSnapshot.GetReportConfigurationId()
+		if req.ReportSnapshot.GetVulnReportFilters() == nil {
```

Comment:
Would it be better to use reportStatus.RunMethod?  Could we ever have a view based report with empty filters?  Even if we can't it still may be better to use the run method.

Reply:
Thats how  view based reports are distinguised in other places